table/locations.go

```diff
@@ -0,0 +1,118 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package table
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/apache/iceberg-go"
+	"github.com/google/uuid"
+)
+
+type LocationProvider interface {
+	NewTableMetadataFileLocation(newVersion int) (string, error)
+	NewMetadataLocation(metadataFileName string) string
+}
+
+type simpleLocationProvider struct {
```

Comment:
```suggestion
type SimpleLocationProvider struct {
```
Shouldn't start this with a capital letter, similar to `LocationProvider`?

Reply:
I didn't want to export the concrete implementation of the interface. I don't want consumers to create instances of `simpleLocationProvider`, I want them to call `LoadLocationProvider` and use the interface instead.